pkg/scaler/scaling_test.go

```diff
@@ -561,17 +512,13 @@ var _ = Describe("Node scaling tests", func() {
 						Expect(found).To(BeTrue())
 						return nil
 					}).Times(2),
-				mockKubeClient.EXPECT().List(gomock.Any(), gomock.Any()).SetArg(1, *nodes),
 				mockKubeClient.EXPECT().List(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).SetArg(1, *upgradeMachines),
 				mockDrainStrategy.EXPECT().Execute(gomock.Any()).Return([]*drain.DrainStrategyResult{}, nil),
 				mockDrainStrategy.EXPECT().HasFailed(gomock.Any()).Return(false, nil),
 				mockKubeClient.EXPECT().List(gomock.Any(), gomock.Any(), []client.ListOption{
 					client.InNamespace(MACHINE_API_NAMESPACE),
 					NotMatchingLabels{LABEL_UPGRADE: "true"},
-				}).SetArg(1, *originalMachineSets),
-				mockKubeClient.EXPECT().List(gomock.Any(), gomock.Any(), []client.ListOption{
-					NotMatchingLabels{machinery.MasterLabel: ""},
-				}).SetArg(1, *masterNodes),
+				}).SetArg(1, *originalMachines),
```

Comment:
The purpose of this test is/was to verify the path of `EnsureScaleDownNodes` that returns `true`. In essence:

- that the function executes a drain strategy 
- that no upgrade-related machines remain in the cluster

But this mock is returning a machine list with 1 in it. Which is not going to get a 'everything is scaled down, return true' path. So really this should be returning an empty machine list.

That probably highlights that we're missing a test in this suite - we want a test exactly like this, but simulating the scenario where the drain strategies have been executed but an upgrade machine is still being returned. And in that scenario the test should expect that `EnsureScaleDownNodes` returns `false`.

Reply:
I have fixed the current block with it expected to return an empty machine list. 

Also added another test in the suite where it will target the scenario where the drain strategies have been executed but an upgrade machine is still being returned and then expects `EnsureScaleDownNodes` returns `false`.